packages/core/src/app/customer/EmailWatermark.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { FunctionComponent, useEffect } from 'react';
+
+import { CheckoutContextProps } from '@bigcommerce/checkout/payment-integration-api';
+import { withCheckout } from '../checkout';
+import { PaymentMethod } from '@bigcommerce/checkout-sdk';
+import { FastlanePrivacySettings, isFastlaneHostWindow } from '@bigcommerce/checkout/paypal-fastlane-integration';
+
+export interface WithCheckoutProps {
+    providerWithCustomCheckout?: string;
+    getPaymentMethod: (methodId: string) => PaymentMethod | undefined;
+}
+
+const EmailWatermark: FunctionComponent<WithCheckoutProps> = (props: WithCheckoutProps) => {
+    const {
+        getPaymentMethod,
+        providerWithCustomCheckout,
+    } = props;
+
+    useEffect(() => {
+        if (
+            providerWithCustomCheckout === 'braintree' ||
+            providerWithCustomCheckout === 'braintreeacceleratedcheckout' ||
+            providerWithCustomCheckout === 'paypalcommerce' ||
+            providerWithCustomCheckout === 'paypalcommerceacceleratedcheckout'
+        ) {
+            const paymentMethod = getPaymentMethod(providerWithCustomCheckout);
+
+            if(
+                paymentMethod?.initializationData?.isFastlanePrivacySettingEnabled &&
+                isFastlaneHostWindow(window)
+            ) {
+                const fastlane = window.braintreeFastlane || window.paypalFastlane;
+                fastlane.FastlaneWatermarkComponent({
+                    includeAdditionalInfo: true,
+                })
+                    .then((result: FastlanePrivacySettings) => {
+                        result.render('#emailWatermark');
+                    });
+            }
+        }
+    }, []);
+
+    return <div className='emailWatermark-container'>
+                <div id='emailWatermark' data-test-id='emailWatermark' />
+           </div>
+}
+
+const mapToCheckoutProps = ({ checkoutState }: CheckoutContextProps): WithCheckoutProps => {
+    return {
+        getPaymentMethod: checkoutState.data.getPaymentMethod,
+        providerWithCustomCheckout: checkoutState.data.getConfig()?.checkoutSettings?.providerWithCustomCheckout || '',
+    };
+}
+
+export default withCheckout(mapToCheckoutProps)(EmailWatermark);
```

Comment:
Can we fix the ⛔️ here?
Also, if possible, can we use checkout context instead of HOC?

Reply:
fixed